src/vcml/models/generic/fbdev.cpp

```diff
@@ -20,6 +20,162 @@ void fbdev::update() {
     }
 }
 
+template <typename T>
+void write_to_ostream(ostream& os, const T* val) {
+    os.write(reinterpret_cast<const char*>(val), sizeof(T));
+}
+
+struct bmp_header {
+    array<char, 2> header; // 0x42 0x4D for BMP (BM)
+    u32 size;              // total size (bytes)
+    u16 reserved_0;
+    u16 reserved_1;
+    u32 offset; // offset of the pixel array
+
+    u32 header_size() { return 14; }
+
+    bmp_header():
+        header({ 'B', 'M' }),
+        size(0),
+        reserved_0(0),
+        reserved_1(0),
+        offset(0) {}
+
+    void write(ostream& out) {
+        write_to_ostream(out, &header[0]);
+        write_to_ostream(out, &header[1]);
+        write_to_ostream(out, &size);
+        write_to_ostream(out, &reserved_0);
+        write_to_ostream(out, &reserved_1);
+        write_to_ostream(out, &offset);
+    }
+};
+struct dib_header {
+    u32 size;           // size of this header (bytes) -> 40
+    i32 width;          // in pixels
+    i32 height;         // in pixels
+    u16 color_planes;   // number of color planes -> 1
+    u16 bits_per_pixel; // typically 1, 4, 5, 16, 24, 32
+    u32 method;         // compression method
+    u32 image_size; // size of the raw bitmap data; a dummy 0 can be given for
```

Comment:
can we avoid the comment line break here? either by shortening the comment or  by putting it above it.